packages/rc-mfa-config/src/components/home/qr-code-verify.tsx

```diff
@@ -104,7 +107,7 @@ export const QrCodeVerify: FC<QrCodeVerifyProps> = ({ refreshAuthenticators, qrC
     },
   })
 
-  useEffect(handleDrawQrCode(modalIsOpen, canvasRef, qrCode), [canvasRef, qrCode, modalIsOpen])
+  useEffect(handleDrawQrCode(modalIsOpen, canvasRef, qrLabel, qrCode), [canvasRef, qrCode, modalIsOpen])
```

Comment:
You need to add connectSession to the dependency array of this useEffect or it will not be called when the value of qrLabel changes. Passing qrLabel will cause it to be called each render because that variable is defined each render so you need to pass in the object it is derived from ie connectSession.

Reply:
Had a read up, makes sense I think. Thanks